pkg/build/strategies/sti/sti.go

```diff
@@ -401,7 +404,7 @@ func (b *STI) Execute(command string, config *api.Config) error {
 		Image:           config.BuilderImage,
 		Stdout:          outWriter,
 		Stderr:          errWriter,
-		PullImage:       config.ForcePull,
+		PullImage:       config.BuilderPullPolicy == api.PullAlways || config.BuilderPullPolicy == api.PullIfNotPresent,
```

Comment:
alternatively if you assume by this point in the code you've already called PullImage (which will have pulled the image if appropriate), then you should always be passing "false" here because the right image (freshly pulled or not) will already exist locally.